migration/2.3/migrationScripts/migrate.js

```diff
@@ -0,0 +1,142 @@
+        },
+        {
+            oldPath: path.join(__dirname, "../server"),
+            newPath: path.join(PROJECT_ROOT, "server"),
+            messageOnComplete: "Copied new server directory into the project."
+        },
+        {
+            oldPath: path.join(__dirname, "../src-built-in"),
+            newPath: path.join(PROJECT_ROOT, "src-built-in"),
+            messageOnComplete: "Copied new presentation components into src-built-in."
+        },
+        {
+            oldPath: path.join(PROJECT_ROOT, "gulpfile.js"),
+            newPath: path.join(PROJECT_ROOT, "gulpfile.js.original"),
+            messageOnComplete: "Copied old gulpfile.",
+            deleteAfterCopy: true
+        },
+        {
+            oldPath: path.join(__dirname, "../gulpfile.js"),
+            newPath: path.join(PROJECT_ROOT, "gulpfile.js"),
+            messageOnComplete: "Copied new gulpfile.",
+        },
+        {
+            oldPath: path.join(PROJECT_ROOT, "build-original/webpack/webpack.files.entries.json"),
+            newPath: path.join(PROJECT_ROOT, "build/webpack/webpack.files.entries.json"),
+            messageOnComplete: "Copied old webpack entries file into the new build directory.",
+        }
+    ],
+        copy,
+        cb);
+}
+
+function copyNodeModules(done) {
+    //If we don't remove node_modules, some folders are not copied over correctly, and the project ends up with multiple versions of react, causing the toolbar to break.
+    log("Removing existing node modules folder.");
+    shelljs.rm('-rf', path.join(__dirname, "../node_modules/"))
+    copy({
+        oldPath: path.join(__dirname, "../node_modules/"),
+        newPath: path.join(PROJECT_ROOT, "node_modules/"),
+        messageOnComplete: "Copied new node modules in."
+    }, done);
+}
+
+function updatePackageFile(done) {
+    let existingPackagePath = path.join(PROJECT_ROOT, "package.json")
+    let existingPackage = require(existingPackagePath);
+    let newPackage = require("../package.json");
+
+    let oldScripts = Object.keys(existingPackage.scripts);
+    let newScripts = Object.keys(newPackage.scripts);
+
+    newScripts.forEach((script) => {
+        if (oldScripts.includes(script)) {
+            log(`Not adding script "${script}" to package.json. Script already exists.`);
+        } else {
+            existingPackage.scripts[script] = newPackage.scripts[script];
+            log(`Adding script "${script}" to package.json.`);
+        }
+    });
+
+    const newPackages = {
+        "uglifyjs-webpack-plugin": "^1.2.3",
+        "webpack": "^2.7.0",
+        "death": "^1.1.0",
+        "cookie-parser": "1.4.3",
+        "babel-minify-webpack-plugin": "^0.3.0"
+    }
+    existingPackage.devDependencies = Object.assign(existingPackage.devDependencies, newPackages);
+    const content = JSON.stringify(existingPackage, null, '\t');
+    fs.writeFileSync(existingPackagePath, content, 'utf8');
+    done();
+}
+
+
+function addVendorBundle(cb) {
```

Comment:
@carterbs let's investigate having the windowHeader load the vendor.bundle.js. That way we won't have to change finsemble-cli immediately.

Reply:
I'll look, but I'm not optimistic. Webpack adds an immediately invoked function into its files that look for the vendor bundle.